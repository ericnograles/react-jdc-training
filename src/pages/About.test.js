import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import About from './About';

describe('<About>', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    let setTitle = jest.fn();
    ReactDOM.render(<About setTitle={setTitle} />, div);
  });

  it('properly shows given text', () => {
    let setTitle = jest.fn();
    const output = shallow(<About setTitle={setTitle} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should set the title of its parent', () => {
    let setTitle = jest.fn();
    const output = mount(<About setTitle={setTitle} />);
    output.find('button').simulate('click');
    expect(setTitle).toHaveBeenCalled();
  });
});
