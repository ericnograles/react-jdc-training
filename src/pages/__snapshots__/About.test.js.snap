// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About> properly shows given text 1`] = `
<ReactComponent
  setTitle={[Function]}
  text="Welcome to /about"
/>
`;
